Mathlib/MeasureTheory/Integral/RieszMarkovKakutani/Real.lean

```diff
@@ -0,0 +1,353 @@
+  · -- The sets `E n` are pairwise disjoint.
+    intro m _ n _ hmn
+    apply Disjoint.preimage
+    simp_rw [mem_preimage, mem_Ioc, disjoint_left]
+    intro x hx
+    rw [mem_setOf_eq, and_assoc] at hx
+    simp_rw [mem_setOf_eq, not_and_or, not_lt, not_le, or_assoc]
+    rcases (by omega : m < n ∨ n < m) with hc | hc
+    · left
+      exact le_trans hx.2.1 (le_tsub_of_add_le_right (hy hc))
+    · right; left
+      exact lt_of_le_of_lt (le_tsub_of_add_le_right (hy hc)) hx.1
+  · -- Upper and lower bound on `f x` follow from the definition of `E n` .
+    intro _ _ hx
+    simp only [mem_inter_iff, mem_preimage, mem_Ioc, E, y] at hx
+    constructor <;> linarith
+  · exact fun _ ↦ (f.1.measurable measurableSet_Ioc).inter measurableSet_closure
+
+omit [LocallyCompactSpace X] in
+/-- Given a set `E`, a function `f : C_c(X, ℝ)`, `0 < ε` and `∀ x ∈ E, f x < c`, there exists an
+open set `V` such that `E ⊆ V` and the sets are similar in measure and `∀ x ∈ V, f x < c`. -/
+lemma exists_open_approx (f : C_c(X, ℝ)) {ε : ℝ} (hε : 0 < ε) (E : Set X) {μ : Content X}
+    (hμ : μ.outerMeasure E ≠ ⊤) (hμ' : MeasurableSet E) {c : ℝ} (hfE : ∀ x ∈ E, f x < c):
+    ∃ (V : Opens X), E ⊆ V ∧ (∀ x ∈ V, f x < c) ∧ μ.measure V ≤ μ.measure E + ENNReal.ofReal ε := by
+  have hε' := ne_of_gt <| Real.toNNReal_pos.mpr hε
+  obtain ⟨V₁ : Opens X, hV₁⟩ := Content.outerMeasure_exists_open μ hμ hε'
+  let V₂ : Opens X := ⟨(f ⁻¹' Iio c), IsOpen.preimage f.1.2 isOpen_Iio⟩
+  use V₁ ⊓ V₂
+  refine ⟨subset_inter hV₁.1 hfE, ?_, ?_⟩
+  · intro x hx
+    suffices ∀ x ∈ V₂.carrier, f x < c from this x (mem_of_mem_inter_right hx)
+    exact fun _ a ↦ a
+  · calc
+      _ ≤ μ.measure V₁ := by simp [measure_mono]
+      _ = μ.outerMeasure V₁ := Content.measure_apply μ (V₁.2.measurableSet)
+      _ ≤ μ.outerMeasure E + ε.toNNReal := hV₁.2
+      _ = _ := by rw [Content.measure_apply μ hμ']; congr;
+
+/-- Choose `N` sufficiently large such that a particular quantity is small. -/
+private lemma exists_nat_large (a' b' : ℝ) {ε : ℝ} (hε : 0 < ε) : ∃ (N : ℕ), 0 < N ∧
+    a' / N * (b' + a' / N) ≤ ε := by
+  have A : Tendsto (fun (N : ℝ) ↦ a' / N * (b' + a' / N)) atTop (𝓝 (0 * (b' + 0))) := by
+    apply Tendsto.mul
+    · exact Tendsto.div_atTop tendsto_const_nhds tendsto_id
+    · exact Tendsto.add tendsto_const_nhds (Tendsto.div_atTop tendsto_const_nhds tendsto_id)
+  have B := A.comp tendsto_natCast_atTop_atTop
+  simp only [add_zero, zero_mul] at B
+  obtain ⟨N, hN, h'N⟩ := (((tendsto_order.1 B).2 _ hε ).and (Ici_mem_atTop 1)).exists
+  exact ⟨N, h'N, hN.le⟩
+
+/-- The main estimate in the proof of the Riesz-Markov-Kakutani: `Λ f` is bounded above by the
+integral of `f` with respect to the `rieszMeasure` associated to `Λ`. -/
+private lemma integral_riesz_aux (f : C_c(X, ℝ)) : Λ f ≤ ∫ x, f x ∂(rieszMeasure hΛ) := by
+  by_cases hX : IsEmpty X
+  · have : Λ f = 0 := by rw [show f = 0 by ext x; refine isEmptyElim x, LinearMap.map_zero Λ]
+    rw [integral_of_isEmpty, this]
```

Comment:
I don't think you are using any more nonemptiness. So you should remove these lines, and unindent the rest of the proof.